uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/service/UniProtEntryService.java

```diff
@@ -86,54 +98,37 @@ public UniProtEntry getByAccession(String accession, String fields) {
         }
     }
 
-    public Stream<UniProtEntry> stream(SearchRequestDTO request) {
-        SolrRequest solrRequest = createSolrRequest(request, false);
-        return storeStreamer.idsToStoreStream(solrRequest);
-    }
-
-    public Stream<String> streamIds(SearchRequestDTO request) {
-        SolrRequest solrRequest = createSolrRequest(request, false);
-        return storeStreamer.idsStream(solrRequest);
+    public Stream<String> streamRDF(SearchRequest searchRequest) {
+        setSizeForDownloadAllIfNeeded(searchRequest);
+        SolrRequest solrRequest = createSolrRequest(searchRequest);
+        return this.storeStreamer.idsToRDFStoreStream(solrRequest);
     }
 
-    public Stream<String> streamRDF(SearchRequestDTO searchRequest) {
-        SolrRequest solrRequest = createSolrRequest(searchRequest, false);
-        return this.storeStreamer.idsToRDFStoreStream(solrRequest);
+    @Override
+    protected SolrRequest createSolrRequest(SearchRequest request) {
+        return createSolrRequest(request, false);
     }
 
-    private SolrRequest createSolrRequest(SearchRequestDTO request, boolean includeFacets) {
-        SolrRequest.SolrRequestBuilder requestBuilder = SolrRequest.builder();
-        requestBuilder.rows(request.getSize());
-        requestBuilder.queryBoosts(queryBoosts);
+    @Override
+    protected SolrRequest createSolrRequest(SearchRequest request, boolean includeFacets) {
```

Comment:
Just thinking that here you could Cast the search request and apply the isoform logic...